integration/ledger/snapshot_test.go

```diff
@@ -1038,59 +1036,69 @@ func invokeWithoutPassingOrdererEndPoint(n *nwo.Network, peer *nwo.Peer, channel
-	ctx, cancelFunc := context.WithTimeout(context.Background(), 10*time.Second)
-	defer cancelFunc()
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
 
-	// get signing identity
 	By("getting the signer for user1 on peer " + peer.ID())
 	signer := n.PeerUserSigner(peer, "User1")
 
-	// create deliver client and delivergroup
-	peerClient := &common.PeerClient{
-		CommonClient: common.CommonClient{
-			GRPCClient: grpcClient(filepath.Join(n.PeerLocalTLSDir(peer), "ca.crt")),
-			Address:    n.PeerAddress(peer, nwo.ListenPort),
+	By("creating the deliver client to peer " + peer.ID())
+	pcc := n.PeerClientConn(peer)
+	defer pcc.Close()
+	df, err := pb.NewDeliverClient(pcc).DeliverFiltered(ctx)
+	Expect(err).NotTo(HaveOccurred())
+	defer df.CloseSend()
+
+	By("starting filtered delivery on peer " + peer.ID())
+	deliverEnvelope, err := protoutil.CreateSignedEnvelope(
+		cb.HeaderType_DELIVER_SEEK_INFO,
+		channelID,
+		signer,
+		&ab.SeekInfo{
+			Behavior: ab.SeekInfo_BLOCK_UNTIL_READY,
+			Start: &ab.SeekPosition{
+				Type: &ab.SeekPosition_Newest{Newest: &ab.SeekNewest{}},
+			},
+			Stop: &ab.SeekPosition{
+				Type: &ab.SeekPosition_Specified{
+					Specified: &ab.SeekSpecified{Number: math.MaxUint64},
+				},
+			},
 		},
-	}
-	deliverClient, err := peerClient.PeerDeliver()
+		0,
+		0,
+	)
 	Expect(err).NotTo(HaveOccurred())
-	dg := ic.NewDeliverGroup([]pb.DeliverClient{deliverClient}, []string{n.PeerAddress(peer, nwo.ListenPort)}, signer, tls.Certificate{}, channelID, txid)
-	err = dg.Connect(ctx)
+	err = df.Send(deliverEnvelope)
 	Expect(err).NotTo(HaveOccurred())
 
-	// create orderer client and broadcast client
-	By("creating orderer client to send transaction to the orderer" + peer.ID())
-	ordererClient := &common.OrdererClient{
-		CommonClient: common.CommonClient{
-			GRPCClient: grpcClient(filepath.Join(n.OrdererLocalTLSDir(orderer), "ca.crt")),
-			Address:    n.OrdererAddress(orderer, nwo.ListenPort),
-		},
-	}
-	bc, err := ordererClient.Broadcast()
+	By("creating orderer client and send transaction to the orderer" + orderer.ID())
+	occ := n.OrdererClientConn(orderer)
+	defer occ.Close()
+	broadcastClient, err := ab.NewAtomicBroadcastClient(occ).Broadcast(context.Background())
 	Expect(err).NotTo(HaveOccurred())
-	broadcastClient := &common.BroadcastGRPCClient{Client: bc}
+
 	err = broadcastClient.Send(env)
 	Expect(err).NotTo(HaveOccurred())
 
-	// wait for deliver event
 	By("waiting for deliver event on peer " + peer.ID())
-	return dg.Wait(ctx)
-}
-
-func grpcClient(tlsRootCertFile string) *comm.GRPCClient {
-	caPEM, res := ioutil.ReadFile(tlsRootCertFile)
-	Expect(res).To(BeNil())
-
-	gClient, err := comm.NewGRPCClient(comm.ClientConfig{
-		Timeout: 3 * time.Second,
-		KaOpts:  comm.DefaultKeepaliveOptions,
-		SecOpts: comm.SecureOptions{
-			UseTLS:        true,
-			ServerRootCAs: [][]byte{caPEM},
-		},
-	})
-	Expect(err).NotTo(HaveOccurred())
-
-	return gClient
+	for {
```

Comment:
What if the deliver event never arrives? Do we need a timeout?

Reply:
https://github.com/hyperledger/fabric/pull/2429/commits/4ce8b576f6ab1828152764d2ac5a9f5c214a6529#r579672777